tests/api_tests/nodes/test_views.py

```diff
@@ -1349,3 +1351,115 @@ def test_return_deleted_private_node_pointer(self):
+
+    def setUp(self):
+        super(TestNodeLogList, self).setUp()
+        self.user = UserFactory.build()
+        password = fake.password()
+        self.logs = NodeLogFactory.build()
+        self.password = password
+        self.user.set_password(password)
+        self.user.save()
+        self.basic_auth = (self.user.username, password)
+
+        self.user_two = UserFactory.build()
+        self.user_two.set_password(self.password)
+        self.user_two.save()
+        self.add = ProjectFactory(is_added=True)
+        self.basic_auth_two = (self.user_two.username, self.password)
+        self.private_project = ProjectFactory(is_public=False, creator=self.user)
+        self.private_url = '/{}nodes/{}/logs/'.format(API_BASE, self.private_project._id)
+        self.public_project = ProjectFactory(is_public=True, creator=self.user)
+        self.public_url = '/{}nodes/{}/logs/'.format(API_BASE, self.public_project._id)
+
+    def assert_datetime_equal(dt1, dt2, allowance=500):
+        assert_less(dt1 - dt2, dt.timedelta(milliseconds=allowance))
+
+    def test_log_create_on_public_project(self):
+        datetime.datetime.strptime('2015-07-28 21:06:34.965114','%Y-%m-%d %H:%M:%S.%f')
+        date_str = "2008-11-10 17:53:59"
+        dt_obj = datetime.datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S")
+        print repr(dt_obj)
+        res = self.app.get(self.public_url)
+        assert_equal(res.status_code, 200)
+        (len(res.json['data']), 1)
+        assert_datetime_equal(datetime.datetime.strptime(res.json['data'][0]['date'], "%Y-%m-%dT%H:%M:%S.%f"),
+                              self.public_project.logs[0].date)
+        assert_equal(res.json['data'][0]['id'], self.public_project.logs[0]._id)
+        assert_equal(res.json['data'][0]['action'], self.public_project.logs[0].action)
+        assert_equal(res.json['data'][0]['version'], self.public_project._version)
+        assert_equal(res.json['data'][0]['name'], self.public_project.logs[0]._name)
+
+    def test_log_create_on_private_project(self):
+        res = self.app.get(self.private_url, auth=self.basic_auth)
+        assert_equal(res.status_code, 200)
+        (len(res.json['data']), 1)
+        assert_datetime_equal(datetime.datetime.strptime(res.json['data'][0]['date'], "%Y-%m-%dT%H:%M:%S.%f"),
+                              self.private_project.logs[0].date)
+        assert_equal(res.json['data'][0]['id'], self.private_project.logs[0]._id)
+        assert_equal(res.json['data'][0]['action'], self.private_project.logs[0].action)
+        assert_equal(res.json['data'][0]['version'], self.private_project.logs[0]._version)
+        assert_equal(res.json['data'][0]['name'], self.private_project.logs[0]._name)
+
+    def test_return_logs_private_project_details(self):
+        res = self.app.get(self.private_url, auth=self.basic_auth_two, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_log_add_contributor_on_public_project(self):
```

Comment:
There's nothing in this test that verifies that, when you add a contributor to the project, the contributor is logged properly.